examples/inverse/plot_morph_VolSourceEstimate.py

```diff
@@ -0,0 +1,282 @@
+    noise_cov = mne.compute_covariance(epochs, tmin=tmin, tmax=0,
+                                       method='shrunk')
+    data_cov = mne.compute_covariance(epochs, tmin=0.04, tmax=0.15,
+                                      method='shrunk')
+    # Read forward model
+    forward = mne.read_forward_solution(fname_fwd)
+
+    # Compute weights of free orientation (vector) beamformer with weight
+    # normalization (neural activity index, NAI). Providing a noise covariance
+    # matrix enables whitening of the data and forward solution. Source
+    # orientation is optimized by setting pick_ori to 'max-power'.
+    # weight_norm can also be set to 'unit-noise-gain'. Source orientation can
+    # also be 'normal' (but only when using a surface-based source space) or
+    # None, which computes a vector beamfomer. Note, however, that not all
+    # combinations of orientation selection and weight normalization are
+    # implemented yet.
+    filters = make_lcmv(evoked.info, forward, data_cov, reg=0.05,
+                        noise_cov=noise_cov, pick_ori='max-power',
+                        weight_norm='nai')
+
+    # Apply this spatial filter to the evoked data.
+    stc = apply_lcmv(evoked, filters, max_ori_out='signed')
+
+    # take absolute values for plotting
+    stc.data[:, :] = np.abs(stc.data)
+
+    # Save result in stc files if desired
+    if fname is not None:
+        stc.save('lcmv-vol')
+
+    # select time window (tmin, tmax) in ms - consider changing for real data
+    # scenario, since those values were chosen to optimize computation time
+    stc.crop(0.087, 0.087)
+
+    src = forward['src']
+
+    # Save result in a 4D nifti file
+    img = mne.save_stc_as_volume(fname, stc, src,
+                                 mri_resolution=True)
+
+    return img, stc, src
+
+
+###############################################################################
+# Save non linear mapping data
+
+def write_morph(fname, morph, overwrite=True):
+    morph_out = dict()
+
+    # dissolve object structure
+    for key, value in morph.items():
+        # save type for order independent decomposition
+        if hasattr(value, '__dict__'):
+            value = value.__dict__
+        morph_out[key] = value
+    if not fname.endswith('.h5'):
+        fname += '.h5'
+    write_hdf5(fname, morph_out, overwrite=overwrite)
+
+
+###############################################################################
+# Load non linear mapping data
+
+def read_morph(fname):
+    if not fname.endswith('.h5'):
+        fname += '.h5'
+    morph_in = read_hdf5(fname)
+
+    morph = dict()
+    # create new instances
+    morph['mapping'] = imwarp.DiffeomorphicMap(None, [])
+    morph['mapping'].__dict__ = morph_in.get('mapping')
+    morph['affine'] = imaffine.AffineMap(None)
+    morph['affine'].__dict__ = morph_in.get('affine')
+
+    morph['affine_reg'] = morph_in.get('affine_reg')
+    morph['domain_shape'] = morph_in.get('domain_shape')
```

Comment:
Same here, populate directly rather than using a bunch of `.get`s

Reply:
above +
In this case it was just handy to "meaningful" address the data... So even if there is a better way to save the data directly without extracting the ``__dict___`` I think it's quite nice to have this morpher dictionary... If we would make ``morpher`` a class, then those could be attributes... but without that I think it's quite handy to have the dictionary structure of the morpher itself... ?